.github/workflows/integration-test.yml

```diff
@@ -1,34 +1,28 @@
-name: Integration Test Suite
+# This workflow is used to run an integration test.
+# The most common use case is that it is triggered by another workflow,
+# such as the Master Merge Queue Suite, or the Integration Comment.
+#
+# It can also be triggered on manual dispatch in CI however.
+# In that use case, an input for the test name needs to be provided.
+# TODO: check if the input is "all" , and run all, without a timeout?
```

Comment:
🤔 I think when I wrote that I was thinking that if we ran all the integration tests together, it could imply overreaching the 30 min (or whatever we settle on) timeout.

Reply:
I don't think it _should_ make a difference. The timeout is per job, which in this case more integration tests are just more parallel jobs, so the same timeout _should_ work?

Reply:
Ah, the critical information here is that this workflow is not the matrix.
It is a single run. And so the enhancement here would be running `make test-integration-all` in a single job.

Reply:
Oh, running serially? I did miss that.

Reply:
The matrix is defined in the master merge queue workflow, and this one is called into run each individually. So technically they aren't serial.